src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -67,6 +77,19 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
                     NetEventSource.Error(null, $"Failed to build chain for {target.Subject}");
                 }
 
+                if (!chainStatus && ChainBuildNeedsTrustedRoot && additionalCertificates != null)
+                {
+                    // Some platforms like Android may not be able to build the chain unless the chain root is trusted.
+                    // We can try to rebuild the chain with making all extra certificates trused.
+                    // We do not try to evaluate trust here, we jsut need to construct the chain so it should not matter.
+                    chain.ChainPolicy.CustomTrustStore.AddRange(additionalCertificates);
+                    chainStatus = chain.Build(target);
+                    if (!chainStatus && NetEventSource.Log.IsEnabled())
+                    {
+                        NetEventSource.Error(null, $"Failed to build chain for {target.Subject}");
```

Comment:
Putting a different message here would help us better understand the logs

```suggestion
                        NetEventSource.Error(null, $"Failed to build chain for {target.Subject} while trusting additional certificates");
```